src/qtoolkit/io/slurm.py

```diff
@@ -117,7 +129,17 @@ def qstate(self) -> QState:
     SlurmState.NODE_FAIL: QState.FAILED,
     SlurmState.OUT_OF_MEMORY: QState.FAILED,
     SlurmState.PENDING: QState.QUEUED,
+    SlurmState.PREEMPTED: QState.FAILED,
+    SlurmState.RESV_DEL_HOLD: QState.QUEUED_HELD,
+    SlurmState.REQUEUE_FED: QState.QUEUED,
+    SlurmState.REQUEUE_HOLD: QState.QUEUED,
```

Comment:
Should these slurm states map to "REQUEUED_HELD" ? Not very clear to me what this REQUEUED_HELD means in the drmaa spec.

Reply:
looking at the descriptions maybe it would be better as `Qstate.REQUEUED`. The `SlurmState.REQUEUE_HOLD` reads is defined as 
> Held job is being requeued.

while the `Qstate.REQUEUED_HELD` is described as 

> REQUEUED HELD: The job was re-queued by the DRM system, and is currently
        placed on hold by the system, the administrator, or the submitting user.

So while the former seems more a job that was held and is now being requeued, the latter describes a state that was requeued, but is now being hold. Which seems somewhat the opposite. Honestly based on the description `QUEUED`, `REQUEUED`, `QUEUED_HELD` and `SUSPENDED` may be fine depending on the nuances of the meaning. I will at least add a comment there, but if you feel any of the other states would be more suitable (indeed `REQUEUED` may be better?) I have nothing against chaning it with any of the above. I would just hope that we will never have to deal with them explicitly...

Reply:
Maybe then REQUEUED fpr the REQUEUED_FED and REQUEUED_HOLD for the REQUEUED_HELD ? (is that what you proposed?). I'm fine with what you decide anyway, but I you suggest, indeed it would be good to add a comment mentioning that the choice for these specific slurm states is not completely clear as the nuances between their definitions is thin